starknet_py/net/http_client.py

```diff
@@ -97,6 +102,28 @@ def handle_rpc_error(result: dict):
     async def handle_request_error(self, request: ClientResponse):
         await basic_error_handle(request)
 
+    async def _warn_if_incompatible_rpc_version(self):
+        if not self._is_spec_version_verified:
+            payload = {
+                "jsonrpc": "2.0",
+                "method": f"starknet_specVersion",
+                "id": 0,
+            }
+
+            res = await self.request(
+                http_method=HttpMethod.POST, address=self.url, payload=payload
+            )
+            spec_version = res["result"]
+
+            if spec_version != EXPECTED_RPC_VERSION:
+                warnings.warn(
+                    f"RPC node with the url {self.url} uses incompatible version {spec_version}. "
+                    f"Expected version: {EXPECTED_RPC_VERSION}",
+                    IncompatibleRPCVersionWarning,
+                    stacklevel=4,
```

Comment:
note: `stacklevel` is set to `4`, so the warning will come from actual file, rather than from http client, e.g. of how it looks below

![image](https://github.com/user-attachments/assets/78fd47eb-5cc7-4730-838c-51fac640d4c4)